### PR TITLE
Remove unused parameters.

### PR DESCRIPTION
**Pull request by corycrean**
- Remove unused parameters.
- Resolve additional clang-tidy warnings.

### Description

CI for https://github.com/ros-planning/moveit2/pull/999 was failing because of unused parameter warnings from clang-tidy.  This PR resolves those warnings.

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by JafarAbdi — APPROVED** [review]

File: moveit_ros/benchmarks/src/BenchmarkOptions.cpp [hunk 1]
```diff
@@ -275,18 +275,22 @@ void BenchmarkOptions::readPlannerConfigs(const rclcpp::Node::SharedPtr& node)
   for (const std::string& pipeline : pipelines)
   {
     std::string pipeline_name;
-    if (!node->get_parameter(np + "." + pipeline + ".name", pipeline_name))
+    std::string pipeline_parameter_name(np);
+    pipeline_parameter_name.append(".");
+    pipeline_parameter_name.append(pipeline);
+    pipeline_parameter_name.append(".name");
```
Comment (tylerjw) [inline moveit_ros/benchmarks/src/BenchmarkOptions.cpp:281]: Minor nit, this string could be a constant if you constructed it on one line.

```suggestion
    const std::string pipeline_parameter_name = np + '.' + pipeline + ".name";
```

**Review by tylerjw — APPROVED** [review]
One tiny nit.  Thank you for fixing these clang-tidy issues.

**Review by vatanaksoytezer — APPROVED** [review]

File: moveit_ros/benchmarks/src/BenchmarkOptions.cpp (hunk 1 above)
Comment (corycrean) [reply moveit_ros/benchmarks/src/BenchmarkOptions.cpp:281]: That's true, but then clang-tidy complains:
```
/root/target_ws/src/moveit2/moveit_ros/benchmarks/src/BenchmarkOptions.cpp:278:58: warning: string concatenation results in allocation of unnecessary temporary strings; consider using 'operator+=' or 'string::append()' instead [performance-inefficient-string-concatenation]
    const std::string pipeline_parameter_name = np + "." + pipeline + ".name";
                                                         ^
```

I could use a lambda to build the string the way I'm doing it here, and still have `pipeline_parameter_name` be const, but I don't know if that would give better performance than your solution (which seems to be clang-tidy's complaint).  What do you think?

Comment (tylerjw) [reply moveit_ros/benchmarks/src/BenchmarkOptions.cpp:281]:
```suggestion
    constexpr auto pipeline_parameter_name = std::string(np).append('.').append(pipeline).append(".name");
```

Maybe that would quiet clang-tidy while still making the variable const without needing the lambda initialization pattern?

Comment (tylerjw) [reply moveit_ros/benchmarks/src/BenchmarkOptions.cpp:281]: I don't actually know if you can make that constexpr or if it just can be const, but if you can try making it constexpr.

Comment (corycrean) [reply moveit_ros/benchmarks/src/BenchmarkOptions.cpp]: That worked (though I couldn't make it constexpr, only const).

**Review by tylerjw — APPROVED** [review]

**Pull request merged**